docs/guides/ui-libraries.md

```diff
@@ -0,0 +1,132 @@
+        <pre>{JSON.stringify(state.values, null, 2)}</pre>
+      </div>
+    </>
+  )
+}
+```
+
+- Initially, we utilize the `useForm` hook from TanStack and destructure the necessary properties. This step is optional; alternatively, you could use `const form = useForm()` if preferred. TypeScript's type inference ensures a smooth experience regardless of the approach.
+- Next, we encapsulate our form elements within the `Provider` component, a critical step for enabling form functionalities. The `Field` component, derived from `useForm`, accepts several properties, such as `validators`. For this demonstration, we focus on two primary properties: `name` and `children`.
+  - The `name` property identifies each `Field`, for instance, `firstName` in our example.
+  - The `children` property leverages the concept of render props, allowing us to integrate components without unnecessary abstractions.
+- TanStack's design relies heavily on render props, providing access to `children` within the `Field` component. This approach is entirely type-safe. When integrating with Mantine components, such as `TextInput`, we selectively destructure properties like `state.value`, `handleChange`, and `handleBlur`. This selective approach is due to the slight differences in types between `TextInput` and the `field` we get in the children.
+- By following these steps, you can seamlessly integrate Mantine components with TanStack Form.
+- This methodology is equally applicable to other components, such as `Checkbox`, ensuring consistent integration across different UI elements.
+
+## Usage with `Material-UI`
+
+The process for integrating Material-UI components is similar. Here's an example using TextField and Checkbox from Material-UI:
+
+```tsx
+        <Field
+            name="lastName"
+            children={({ state, handleChange, handleBlur }) => {
+              return (
+                <TextField
+                  id="filled-basic"
+                  label="Filled"
+                  variant="filled"
+                  defaultValue={state.value}
+                  onChange={(e) => handleChange(e.target.value)}
+                  onBlur={handleBlur}
+                  placeholder="Enter your last name"
+                />
+              );
+            }}
+          />
+
+           <Field
+            name="isMuiCheckBox"
+            children={({ state, handleChange, handleBlur }) => {
+              return (
+                <MuiCheckbox
+                  onChange={(e) => handleChange(e.target.checked)}
+                  onBlur={handleBlur}
+                  checked={state.value}
+                />
+              );
+            }}
+          />
+
+```
+
+- The integration approach is the same as with Mantine.
+- The primary difference lies in the specific Material-UI component properties and styling options.
+
+# Sandbox example with `Material` and `Mantine`
+
+[UI Library intergration](https://codesandbox.io/p/devbox/demo-form-query-x7w2gz)
```

Comment:
I don't think this should be needed. Instead, make a directory called "ui-libraries" in examples in this repo.

Reply:
Alright done